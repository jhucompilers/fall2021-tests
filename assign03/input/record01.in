PROGRAM person;
  TYPE Person = RECORD
    name: ARRAY 10 OF CHAR;
    age: INTEGER;
  END;
  VAR p: Person;

BEGIN
  READ p.name;
  READ p.age;

  IF p.age >= 18 THEN
    WRITE 1;
  ELSE
    WRITE 0;
  END;
END.
