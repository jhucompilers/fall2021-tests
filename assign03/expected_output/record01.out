1,VAR,name,ARRAY 10 OF CHAR
1,VAR,age,INTEGER
0,TYPE,Person,RECORD (ARRAY 10 OF CHAR x INTEGER)
0,VAR,p,RECORD (ARRAY 10 OF CHAR x INTEGER)
